tests/_plugins/ui/_impl/test_data_explorer.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2024 Marimo. All rights reserved.
+from __future__ import annotations
+
+import pytest
+
+from marimo._dependencies.dependencies import DependencyManager
+from marimo._plugins.ui._impl import data_explorer
+from marimo._runtime.runtime import ExecutionContext
+from tests.conftest import MockedKernel
+
+HAS_DEPS = DependencyManager.has_pandas()
+
+
+@pytest.mark.skipif(not HAS_DEPS, reason="optional dependencies not installed")
+def test_data_explorer() -> None:
+    import pandas as pd
+
+    # Create kernel and give the execution context an existing cell
+    mocked = MockedKernel()
+    mocked.k.execution_context = ExecutionContext("test_cell_id", False)
```

Comment:
Oh creating a `MockedKernel` like this can lead to problems because it won't get cleaned up. I added a fixture for this called `executing_kernel`.

`def test_data_explorer(executing_kernel: Kernel) -> None:`